### PR TITLE
Retry HTTP 520 errors

### PR DESCRIPTION
**Pull request by Gallaecio**
Fixes https://github.com/scrapy-plugins/scrapy-zyte-api/issues/25

### CODE REVIEW

File: zyte_api/aio/retry.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def _is_network_error(exc: BaseException) -> bool:
 
 
 def _is_throttling_error(exc: BaseException) -> bool:
-    return isinstance(exc, RequestError) and exc.status in (429, 503)
+    return isinstance(exc, RequestError) and exc.status in (429, 503, 520)
```
Comment (BurnzZ) [inline zyte_api/aio/retry.py:55]: For these errs, we'll need to follow the value specified in the `Retry-After` header.

What do you think about updating https://github.com/zytedata/python-zyte-api/blob/main/zyte_api/aio/retry.py#L71-L81 to check the headers?

Although we could do that in a separate PR to prevent blocking this quick fix.

**Review by BurnzZ — APPROVED** [review]

File: zyte_api/aio/retry.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def _is_network_error(exc: BaseException) -> bool:
...
```
Comment (kmike) [inline zyte_api/aio/retry.py:54]: 520 is not a throttling error, I think it's a real error

Comment (kmike) [reply zyte_api/aio/retry.py:54]: It might require different retry policy, and also we may want to have it configurable

Comment (Gallaecio) [reply zyte_api/aio/retry.py:54]: It is described as a temporary download error in the docs, but I know of at least 1 URL that consistently gets this error, so yes, retrying forever is probably not a good option.

Comment (kmike) [reply zyte_api/aio/retry.py:54]: FTR, zyte-autoextract handled a similar case with [max_query_error_retries](https://github.com/zytedata/zyte-autoextract/blob/696421817eefca8720395249551c85fd4c838238/autoextract/aio/client.py#L164) option.
